### PR TITLE
Feature/reader x posts

### PR DESCRIPTION
**Pull request by aerych**
This PR adds new styling and behavior for cross-posts that we receive in certain reader feeds. 

To test:
- View a feed that contains cross posts. 
- Confirm that the cross post cell content matches the appearance of cross post cells on the web. (Note that the author name may appear slightly different and this is expected.)
- Confirm that tapping the cross post cell takes you to the source post's detail, not the detail of the cross post. 

Needs review: @kwonye

### CODE REVIEW

File: WordPress/Classes/Services/ReaderPostService.m [hunk 1]
```diff
@@ -905,6 +905,21 @@ - (ReaderPost *)createOrReplaceFromRemotePost:(RemoteReaderPost *)remotePost for
     post.isLikesEnabled = remotePost.isLikesEnabled;
     post.isSiteBlocked = NO;
 
+    if (remotePost.xpostMeta) {
+        if (!post.xpostMeta) {
+            ReaderXPostMeta *meta = (ReaderXPostMeta *)[NSEntityDescription insertNewObjectForEntityForName:@"ReaderXPostMeta"
```
Comment (kwonye) [inline WordPress/Classes/Services/ReaderPostService.m]: @"ReaderXPostMeta" can be a const

**Comment by kwonye** [comment]
One thing that concerns me is the multiple spellings of X-Post. There's:

1. XPost (eg. `ReaderXPostCell`, `Cards.xPostSubtitleFontSize`)
2. Xpost (eg. `post.xpostMeta`)
3. CrossPost (eg. isCrossPost)

Considering the actual user-facing text is 'cross-post', can we make all references `CrossPost`?

File: WordPress/Classes/Services/ReaderPostService.m (hunk 1 above)
Comment (aerych) [inline WordPress/Classes/Services/ReaderPostService.m]: Oops! I was being lazy when I was building.  A const would work but I might go with `[ReaderXPostMeta className]`

Comment (kwonye) [inline WordPress/Classes/Services/ReaderPostService.m]: > I might go with [ReaderXPostMeta className]

:+1:

**Comment by aerych** [comment]
> Considering the actual user-facing text is 'cross-post', can we make all references `CrossPost`

Works for me :)

**Comment by aerych** [comment]
Ready for another round @kwonye :)

**Comment by kwonye** [comment]
Works great! I noticed there are still some lingering `xpost`s, but they're all connected to `CrossPost` so I trust your judgement on them.

Tested online/offline functionality and works great! :shipit:

**Comment by aerych** [comment]
> I noticed there are still some lingering xposts

Ah... the ones for the REST API keys (where fields are actually named "xpost") were intentional but I missed the ones in WPStyleGuide! I'll update those real quick before I merge. 

Thanks @kwonye !

**Comment by kwonye** [comment]
Thank you! I love this new feature btw. Hated popping into the webviews in the reader.

**Pull request merged**